### ISSUE TITLE
Ports hard-coded in wallet/src/types/rs

### ISSUE THREAD

**Issue by photis** [mimblewimble/grin#441]
On line 174 : `api_listen_port: “13415".to_string(),`
On line 175: `check_node_api_http_addr: "http://127.0.0.1:13413".to_string(),`
These ports are hard-coded and no matter what values they are set to in grin.toml, they don't change.

Line 174 manifests in  
	`INFO Starting the Grin wallet receiving daemon at 0.0.0.0:13415...` when wallet started

Line 175 manifests in
	`WARNING - Showing local data only - Wallet was unable to contact a node to update and verify the outputs shown here.` when wallet info or outputs is requested and the server api port is set to a different value in grin.toml in the _**server**_ subdirectory, even when the grin.toml file in the **_wallet_** subdirectory sets the server api port to that same value.

**Comment by heunglee** [comment]
Those are default values. On line 181, there is the configuration function.

**Comment by photis** [comment]
Sorry, what I'm saying is that these default values never change, they don't take the value of the settings in the .toml files. I have 2 identical grin.toml files in each of **_server_** and **_wallet_** sub-directory, they both have the same settings for the ports (different from the default 13413-13415 range). It doesn't work. When I hard-code those port values on lines 174-175 in wallet/src/types.rs with the same values as used in my grin.toml files, everything works as it should.

**Comment by ignopeverell** [comment]
Fixed in 900e5cd, please check the comments for additional context.

Now to add wallet config you will need a specific wallet configuration block like so:

```
[wallet]
api_listen_interface: "x.x.x.x"
api_listen_port: 1377
```

Hopefully we can simplify all this in the near-future...

**Comment by heunglee** [comment]
@ignopeverell I am confused a little bit. 

The comments in default of WalletConfig:
 // The api interface/ip_address that this api server (i.e. this wallet) will run
 // by default this is 127.0.0.1 (**and will not accept connections from external clients**)

I presume the default of localhost is for security reason. Allowing to configure to api_listen_interface IP other than localhost can allow a external client, can't it?
